app/controllers/publish/providers_controller.rb

```diff
@@ -39,8 +61,38 @@ def update
       end
     end
 
+    def search
+      skip_authorization
+
+      provider_query = params[:query]
+
+      if provider_query.blank?
+        flash[:error] = { id: "provider-error", message: "Name or provider code" }
+        return redirect_to publish_root_path
+      end
+
+      provider_code = provider_query
+                        .split
+                        .last
+                        .gsub(/[()]/, "")
+
+      redirect_to publish_provider_path(provider_code)
+    end
+
   private
 
+    def provider
```

Comment:
Are we able to re-use this from the parent controller? https://github.com/DFE-Digital/teacher-training-api/blob/master/app/controllers/publish/publish_controller.rb#L11

Reply:
yep

Reply:
Actually it causes a pundit error in the tests